### PR TITLE
[BUGFIX] Instantiate new page repository for backend context

### LINKED ISSUE #1066: vhs in backend partially broken in 3.0.0
I use the DCE extension and defined custom backend views for the content elements.
For example, using the v:page.info viewhelper, I get the title of a page:
```
{namespace v=FluidTYPO3\Vhs\ViewHelpers}
{v:page.info(pageUid: '1', field: 'title')}
```

This no longer works in 3.0.0, despite working in 2.4.0 without any issues.
When saving a DCE element that has such a viewhelper, it will save it just fine (frontend shows the updated element), but it will result in an error for the backend view. PHP Error log shows this:
> PHP Fatal error:  __clone method called on non-object in /var/www/typo3conf/ext/vhs/Classes/Service/PageService.php on line 190

The line: https://github.com/FluidTYPO3/vhs/blob/development/Classes/Service/PageService.php#L190

I am running 7.6.9 LTS with DCE 1.3.6 with PHP5

### PR DESCRIPTION
**Pull request by bjo3rnf**
Fixes #1066

### CODE REVIEW

File: Classes/Service/PageService.php [hunk 1]
```diff
@@ -172,7 +177,7 @@ public function hidePageForLanguageUid($page = null, $languageUid = -1, $normalW
         $hideIfDefaultLanguage = (boolean) GeneralUtility::hideIfDefaultLanguage($l18nCfg);
         $pageOverlay = [];
         if (0 !== $languageUid) {
-            $pageOverlay¨ = $GLOBALS['TSFE']->sys_page->getPageOverlay($pageUid, $languageUid);
```
Comment (bjo3rnf) [inline Classes/Service/PageService.php]: Oups! What is this?

File: Classes/Service/PageService.php [hunk 2]
```diff
@@ -187,10 +192,24 @@ public function hidePageForLanguageUid($page = null, $languageUid = -1, $normalW
...
         return clone $GLOBALS['TSFE']->sys_page;
     }
 
     /**
+     * @return PageRepository
+     */
+    protected function getPageRepositoryForBackendContext()
+    {
+        if (static::$backendPageRepository === null) {
+            static::$backendPageRepository = GeneralUtility::makeInstance('TYPO3\\CMS\\Frontend\\Page\\PageRepository');
```
Comment (cedricziel) [inline Classes/Service/PageService.php]: Shorthand class reference please

File: Classes/Service/PageService.php [hunk 3]
```diff
@@ -42,6 +42,11 @@ class PageService implements SingletonInterface
     protected static $cachedRootlines = [];
 
     /**
+     * @var \TYPO3\CMS\Frontend\Page\PageRepository
```
Comment (cedricziel) [inline Classes/Service/PageService.php]: Shorthand class reference please

Comment (bjo3rnf) [inline Classes/Service/PageService.php]: My apologies.

**Pull request merged**